.cloudbuild/scripts/can-skip/main.go

```diff
@@ -0,0 +1,194 @@
+		err = touch(skipFile)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("Non-docs changes detected. Tests will run as normal.")
+	}
+
+	return nil
+}
+
+// hasOnlyDocChanges examines a collection fo changes to check that there are
+// only documentation related changes. Doc changes are assumed to be related
+// to files with given prefices or extensions.
+func hasOnlyDocChanges(changes object.Changes) bool {
+	for _, change := range changes {
+		path := changePath(change)
+		if path == "" ||
+			strings.HasPrefix(path, "docs/") ||
+			strings.HasSuffix(path, ".mdx") ||
+			strings.HasSuffix(path, ".md") {
+			continue
+		}
+
+		return false
+	}
+
+	return true
+}
+
+// getChanges resolves the head of target branch and compares the trees at the
+// the target branch and the supplied commit SHA.
+func getChanges(repo *git.Repository, targetBranch, commit string) (object.Changes, error) {
+	fmt.Printf("Getting worktree for target branch %s\n", targetBranch)
+	targetTree, err := getBranchTree(repo, targetBranch)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting target branch worktree: %w", err)
+	}
+
+	fmt.Printf("Getting worktree for commit %s\n", commit)
+	commitTree, err := getCommitTree(repo, commit)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting target branch worktree: %w", err)
+	}
+
+	fmt.Printf("Comparing commit %s with target %s\n", commit, targetBranch)
+	changes, err := targetTree.Diff(commitTree)
+	if err != nil {
+		return nil, fmt.Errorf("failed diffing target branch and latest commit: %w", err)
+	}
+
+	fmt.Printf("There are %d changes:\n", changes.Len())
+	return changes, nil
+}
+
+func getBranchTree(repo *git.Repository, name string) (*object.Tree, error) {
+	refname := plumbing.NewRemoteReferenceName("origin", name)
+	fmt.Printf("Treating target branch as: %s\n", refname)
+	sha, err := repo.ResolveRevision(plumbing.Revision(refname))
+	if err != nil {
+		return nil, fmt.Errorf("failed resolving branch %q: %w", name, err)
+	}
+
+	fmt.Printf("Branch %s resolves to %s\n", name, sha)
+
+	return getTreeForSHA(repo, *sha)
+}
+
+func getCommitTree(repo *git.Repository, ref string) (*object.Tree, error) {
+	sha, err := repo.ResolveRevision(plumbing.Revision(ref))
+	if err != nil {
+		return nil, fmt.Errorf("failed resolving commit %q: %w", ref, err)
+	}
+
+	fmt.Printf("Ref %s resolves to %s\n", ref, sha)
+
+	return getTreeForSHA(repo, *sha)
+}
+
+func getTreeForSHA(repo *git.Repository, sha plumbing.Hash) (*object.Tree, error) {
+	commit, err := repo.CommitObject(sha)
+	if err != nil {
+		return nil, fmt.Errorf("failed extracting commit %q: %w", sha, err)
+	}
+
+	tree, err := commit.Tree()
+	if err != nil {
+		return nil, fmt.Errorf("failed getting commit %q worktree: %w", sha, err)
+	}
+
+	return tree, nil
+}
+
+func touch(filename string) error {
+	f, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed touching file %s: %w", filename, err)
```

Comment:
`trace.Wrap` here and elsewhere.

Reply:
No worries. Before this was a full-on module I was trying to avoid pulling in extra packages and using the built-in go 1.13+ error wrapping. This doesn't matter as much any more, as I'm already pulling in a bunch of dependencies - and consistency with the rest of the codebase is more important.